qa-tests-backend/src/test/groovy/ExternalNetworkSourcesTest.groovy

```diff
@@ -198,34 +214,62 @@ class ExternalNetworkSourcesTest extends BaseSpecification {
         }
 
         and:
-        "Verify edge from deployment to supernet exists in older network graph"
+        "Verify edge from deployment 'external-connection' to supernet external source '$externalSource30Name' exists" +
+            " in the network graph for last 60 minutes"
         withRetry(4, 30) {
             assert NetworkGraphUtil.checkForEdge(
-                    deploymentUid,
-                    externalSource30ID,
-                    Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60*60).build())
+                deploymentUid,
+                externalSource30ID,
+                Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60 * 60).build())
         }
 
         and:
-        "Verify no edge from deployment to supernet exists in recent network graph"
-        withRetry(4, 30) {
+        "Verify no edge exists from 'external-connection' to supernet external source '$externalSource30Name' " +
+            "in the network graph for last 60 seconds"
+        withRetry(20, 30) {
+            // We need to wait for at least (i.e., the sum of all the following):
+            // - another enrichment to happen - at least 30s.
+            // - old connection being closed by Collector - at least 5 min with the default afterglow setting.
+            // - the time-scope of the network graph - here 60s.
+            // Waiting for 6 min and 30s is a minimum here, however the edge may disappear sooner.
+            // Manually observing this test-case confirmed that there are two updates for "externalSource30ID"
+            // sent to Central, the second exactly 5min30s after the first one.
+            // We set the retries to cover 10 minutes to account for unpredictable issues.
             assert verifyNoEdge(
-                    deploymentUid,
-                    externalSource30ID,
-                    Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60).build())
+                deploymentUid,
+                externalSource30ID,
+                Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60).build())
         }
 
         cleanup:
         deleteNetworkEntity(externalSource30ID)
         deleteNetworkEntity(externalSource31ID)
     }
 
+    private static List<String> getConflictingCIDRs(List<String> cidrs) {
+        def clusterId = ClusterService.getClusterId()
+        assert clusterId
+        def request = NetworkGraphServiceOuterClass.GetExternalNetworkEntitiesRequest
+            .newBuilder()
+            .setClusterId(clusterId)
+            .build()
+        def response = NetworkGraphService.getNetworkGraphClient().getExternalNetworkEntities(request)
+
+        def existingCidrs = response.getEntitiesList().findAll {
+            it.getInfo().hasExternalSource()
+        }.collect {
+            it.getInfo().getExternalSource().cidr
+        } as Set
+
+        return cidrs.findAll { cidr -> cidr in existingCidrs }
```

Comment:
nit: If this function accepts and returns set instead of List then we can use guava code to get intersections
```suggestion
        return Sets.intersection(cidrs, existingCidrs)
```
```
import com.google.common.collect.Sets
```

Reply:
Here is my attempt to change that to set: [2338a1f](https://github.com/stackrox/stackrox/pull/15156/commits/2338a1f77c44828dafadd2ae9f087bbd28894ffe) 
It does cast few times, does that look ok?